pkg/runtime/conversion_generator.go

```diff
@@ -86,9 +87,15 @@ func (g *conversionGenerator) AddImport(pkg string) string {
 	return g.addImportByPath(pkg)
 }
 
-func (g *conversionGenerator) GenerateConversionsForType(version string, reflection reflect.Type) error {
+func (g *conversionGenerator) GenerateConversionsForType(gvString string, reflection reflect.Type) error {
+	gv, err := unversioned.ParseGroupVersion(gvString)
+	if err != nil {
+		return err
+	}
+
 	kind := reflection.Name()
-	internalObj, err := g.scheme.NewObject(g.scheme.InternalVersion, kind)
+	// TODO this is equivalent to what it did before, but it needs to be fixed for the proper group
+	internalObj, err := g.scheme.NewObject(g.scheme.InternalVersions[gv.Group].String(), kind)
```

Comment:
right, not a nil check, but a presence check, otherwise you get the default zero value for the map (`GroupVersion{}`) and pass "" to `NewObject`, right?